# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/cookiestore/CookieChangeEvent.h
# Commit: 5e078f860dfb
# Full Hash: 5e078f860dfb9368bd747e4a8c53d07c7e8809ee
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2024-09-13 04:28:43
# Regressor Bug: 1475599
# File Overlap Count: 1
# Description:
#   Bug 1475599 - part 2 - CookieStore API - IPDL, r=edgul,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D215145
# ==============================================================================

diff -r d55b6525d7b2 -r 5e078f860dfb dom/cookiestore/CookieChangeEvent.h
--- a/dom/cookiestore/CookieChangeEvent.h	Thu Sep 12 16:48:18 2024 +0000
+++ b/dom/cookiestore/CookieChangeEvent.h	Thu Sep 12 16:48:19 2024 +0000
@@ -29,6 +29,12 @@
 
   void GetDeleted(nsTArray<CookieListItem>& aList) const;
 
+  static already_AddRefed<CookieChangeEvent> CreateForChangedCookie(
+      EventTarget* aEventTarget, const CookieListItem& aItem);
+
+  static already_AddRefed<CookieChangeEvent> CreateForDeletedCookie(
+      EventTarget* aEventTarget, const CookieListItem& aItem);
+
  private:
   ~CookieChangeEvent();
 